components/nimbus/src/stateless/cirrus_client.rs

```diff
@@ -133,27 +135,39 @@ impl CirrusClient {
         let nimbus_id = Uuid::new_v4();
         let available_randomization_units =
             AvailableRandomizationUnits::with_client_id(client_id.as_str());
-        let th = NimbusTargetingHelper::new(targeting_attributes);
+        let ta = TargetingAttributes::new(self.app_context.clone(), request_context);
+        let th = NimbusTargetingHelper::new(ta);
         let enrollments_evolver =
             EnrollmentsEvolver::new(&nimbus_id, &available_randomization_units, &th);
+        let state = self.state.lock().unwrap();
 
         let (enrollments, events) = enrollments_evolver
             .evolve_enrollments::<EnrolledFeatureConfig>(
                 is_user_participating,
                 Default::default(),
-                next_experiments,
+                &state.experiments,
                 prev_enrollments,
             )?;
 
         let enrolled_feature_config_map =
-            map_features_by_feature_id(&enrollments, next_experiments);
+            map_features_by_feature_id(&enrollments, &state.experiments);
```

Comment:
IIRC, you had something to unpack the feature id --> raw feature JSON, rather than passing the `EnrolledFeatureConfig` back. What happened to that? Did I imagine that? 😊

Reply:
So I don't _think_ I had ever had something like that, but I know we talked about something that could take a `FeatureConfig`, validate it against a FeatureManifest, and merge it into the feature default JSON object.

That has been done over in the FML library, as a part of this PR: https://github.com/mozilla/application-services/pull/5557/files#diff-c664e1568e3d1b6098e9fbc6422dc27647782af7afa92b0e957e3e191d82f2e4R139-R174 (it's still a draft, just wanted to get it up so Jared/Yashika could review the bindings)